### ISSUE TITLE
Visualizing ChIPSeqData 

### ISSUE THREAD

**Issue by sengoku93** [deeptools/deepTools#689]
Hi,

I am currently trying to generate peakcentered ChIP seq heatmaps. However, I'm having trouble generating these heatmaps because the resulting heat map  has a single value per gene (one specific color) rather than several values per gene dependent on bin size (therefore multiple colors per row). Below are the multiple commands I tried (all resulting in similar heatmap that is attached, cropped image)(I also attached a representative example of the type of heatmap I'd like to generate, called represenative heatmap):

**bamCoverage:**

bamCoverage -b input.bam -o output.bw  
bamCoverage -b input.bam -o output.bw --centerReads

**Compute Matrix:**

computeMatrix reference-point -S input.bw  -R gencode.v19.annotation.gtf  --referencePoint center --sortRegions keep -a 1000 -b 1000 -bs 50 -o output.gz

computeMatrix reference-point -S input.bw  -R gencode.v19.annotation.gtf  --referencePoint center --sortRegions keep -a 1000 -b 1000 -bs 10 -o output.gz

**plotHeatmap:**

plotHeatmap -m p63 input.gz  -out ouput_v19.png --sortRegions no --colorMap 'RdBu'

Any help would be appreciated! Thank you!

-sengoku

![10bp_v19_cropped](https://user-images.githubusercontent.com/37084775/38001840-1789406a-31fd-11e8-870b-1fa862213aea.png)

<img width="188" alt="representative heatmap" src="https://user-images.githubusercontent.com/37084775/38001996-dd943f62-31fd-11e8-9259-68853b8d15b5.png">

**Comment by dpryan79** [comment]
What version of deepTools are you using? This behavior is indeed very odd.

Devon

--
Devon Ryan, Ph.D.
Email: dpryan@dpryan.com
Data Manager/Bioinformatician
Max Planck Institute of Immunobiology and Epigenetics
Stübeweg 51
79108 Freiburg
Germany
<devon.ryan@dzne.de>

**Comment by sengoku93** [comment]
Hi.

deepTools 3.0.1 installed in conda

**Comment by sengoku93** [comment]
This is also in python 2.7, I will try using a conda environment with most up to date python version. Thank you

**Comment by sengoku93** [comment]
I have tried bamCoverage using updated python version 3.5.4. An error occurs:

Segmentation fault: 11

**Comment by dpryan79** [comment]
I suspect that your conda environment is somehow broken. Can you (likely privately) make your input.bam file available? I can then have a look.

**Comment by sengoku93** [comment]
can i send via dropbox? I have free acount on github, not sure if I can share files with you/don't how to upload into repository

**Comment by dpryan79** [comment]
Yes, dropbox would be preferable. You can send the invitation to dpryan79@gmail.com

**Comment by sengoku93** [comment]
Sent. Thank you for taking your time to help me. Very much appreciated. 

-sengoku

**Comment by dpryan79** [comment]
The plots accurately reflect the distribution of signal in your samples. Your data is mostly uniformly distributed across the genic and intergenic regions. There is a small bit of enrichment at the transcription start site, which you can see with:

    computeMatrix reference-point -a 1000 -b 1000 -S input.bw -R gencode.v19.annotation.gtf -o output.mat.gz
    plotHeatmap -m output.mat.gz -o output.png

**Comment by sengoku93** [comment]
Is this for a peak centered heat map? I am not concerned with enrichment near the tss. I want to be able to generate a heatmap with all peaks calls centered at position 0 that is sorted in a manner that I can use to compare it to other heatmaps I'd generate. Would you be able to point me to software in that direction? Below is a reference (Figure 3C) for the type of heat map I'd like to generate. 

Jeselsohn, Rinath, et al. "Allele-specific chromatin recruitment and therapeutic vulnerabilities of ESR1 activating mutations." Cancer cell 33.2 (2018): 173-186.


Thank you for helping me. 

-sengoku

**Comment by dpryan79** [comment]
No, but you were inputting genes not peaks. deepTools was likely used to produce the heatmaps you linked to, you just need to feed actual peaks into it rather than genes.

**Comment by sengoku93** [comment]
Therefore, instead of the gencodev19 I would use a bed file generate from Macs2 peak calling? thank you

**Comment by dpryan79** [comment]
Correct

**Comment by sengoku93** [comment]
Does computeMatrix accept bed files for -R input? Thank you!

**Comment by dpryan79** [comment]
It does
